Create Windows Configuration Recommendations page (#2304)
nuxeo/doc.nuxeo.com-content: src/client-apps/nuxeo-drive/nuxeo-drive-faq/nuxeo-drive-architecture-recommendations-windows.md

---
title: Nuxeo Drive Configuration Requirements on Windows
review:
  comment: ''
  date: '2024-10-05'
  status: ok
labels:
  - nuxeo-drive
  - mlumeau
  - multiexcerpt
  - excerpt

>   - lts2023-ok

Reviewer: We can also specify the Nuxeo Drive version for which this requirements is recommended

toc: true
tree_item_index: 100
---

## Before You Install

### Supported Windows Version 

- Windows >= 10, both 32 and 64 bits

### Nuxeo Drive Version 

All Nuxeo Drive versions are compatible with Windows. We recommend to use the latest stable version to benefit from the latest bugfixes and improvements.
  
### CPU and RAM Requirements 

Resources consumption depends mostly on the third-party application used for file editing as well as the size of the file. 
### Requirements for Direct Edit

> Nuxeo Drive is tested against Microsoft Office 365 and is therefore our recommendation. Direct edit can also be used with alternative Office suites, but has not been tested against them.

Reviewer 2: ```suggestion
Nuxeo Drive is tested against Microsoft Office 365 and is therefore our recommendation. Direct edit can also be used with alternative office suites, but has not been tested against them.
```

To use Direct Edit, you should allocate at least **500MB of free memory** for Nuxeo Drive on top of Microsoft Office consumption for direct edit usage or at least **twice the file size** for larger files.

### Permissions 

Administrator rights are required on the machine while installing or uninstalling the application. Upgrade is similar to installation, so a similar level of access is required.  

### Dependencies 

> Starting from Nuxeo-drive version 5.5.0 => Python v3.9.5 is compatible and required to be installed separately in the local machine. 

Reviewer: Python 3.9.5 is supported from 5.2.0
[NXDRIVE-2629](https://jira.nuxeo.com/browse/NXDRIVE-2629): Upgrade from Python 3.9.4 to 3.9.5

> Starting from Nuxeo-drive version 5.2.0 => Python v3.9.5 is compatible and required to be installed separately in the local machine. 

Reviewer 2: ```suggestion
Starting from Nuxeo-drive version 5.2.0, Python `v3.9.5` is the required version to be installed separately in the local machine. Please make sure to use this version specifically as bugs could happen even with a patch upgrade.
```

Reviewer 2: We need to make that very very clear because even a patch update to 3.9.12 contains breaking changes that will cause issues with drive

---
title: Nuxeo Drive Configuration Requirements on Windows
review:
  comment: ''
  date: '2024-10-05'
  status: ok
labels:
  - nuxeo-drive
  - mlumeau
  - multiexcerpt
  - excerpt
  - lts2023-ok
toc: true
tree_item_index: 100
---

## Before You Install

### Supported Windows Version 

- Windows >= 10, both 32 and 64 bits

### Nuxeo Drive Version 

> - Use Latest Nuxeo Drive version 

Reviewer 2: ```suggestion
All Nuxeo Drive versions are compatible with Windows. We recommend to use the latest stable version to benefit from the latest bugfixes and improvements.
```

  
### CPU and RAM Requirements 

> It depends on the third-party application used for file editing as well as the size of the file. Nuxeo Drive is tested against Microsoft Office 365 and we recommend using it.

Reviewer 2: ```suggestion
Resources consumption depends mostly on the third-party application used for file editing as well as the size of the file. 
```

> To use Direct Edit, you should allocate at least **500MB of free memory** for Nuxeo Drive on top of Microsoft Office consumption for direct edit usage or at least **twice the file size** for larger files.

Reviewer 2: ```suggestion
### Requirements for Direct Edit
Nuxeo Drive is tested against Microsoft Office 365 and is therefore our recommendation. Direct edit can also be used with alternative Office suites, but has not been tested against them.

To use Direct Edit, you should allocate at least **500MB of free memory** for Nuxeo Drive on top of Microsoft Office consumption for direct edit usage or at least **twice the file size** for larger files.
```

### Permissions 

> Administrator rights are required on the machine while installing or uninstalling the application. Upgrade is similar to installation, so similar level of access is required.  

Reviewer 2: ```suggestion
Administrator rights are required on the machine while installing or uninstalling the application. Upgrade is similar to installation, so a similar level of access is required.  
```

### Dependencies 

> - Nuxeo-drive version 5.5.0 => Python v3.9.5 is compatible and required to be installed separately in the local machine. 

Reviewer 2: Why mention a specific drive version here? Should be more generic or mention "starting from" at least.

> => Make sure that no other software overrides the supported python version. 

Reviewer 2: ```suggestion
Make sure that no other software overrides the supported python version. 
```

> Nuxeo Drive embeds its own dependencies, you should ensure that you do not override these dependencies system wide to prevent issues. 

Reviewer 2: ```suggestion
Although Nuxeo Drive embeds its own dependencies, you should ensure that you do not override these dependencies system wide to prevent issues. 
```

We recommend using Microsoft Office 365 or the latest version for Direct Edit.  

## Installation 

### Localization and Permission

It is required to install Nuxeo Drive in the local Users folder. We do not support installing Nuxeo Drive in a different folder or folder type (e.g., redirected folder, network folder).

You need at least Read and Write access to the local Users folder. 

Info: 
In order to show the icon overlay on the explorer on Windows platform, user needs the permission to update the registry editor. 

### Account Information 

> Once Nuxeo Drive installed all the information related to the account, (e.g: Account configuration, database, logs, etc.) will be stored in a folder named `.nuxeo-drive` stored inside the user profile folder. No specific permission are required to store this information. 

Reviewer 2: ```suggestion
Once Nuxeo Drive is installed, all the information related to the account, (e.g: Account configuration, database, logs, etc.) will be stored in a folder named `.nuxeo-drive` stored inside the user profile folder. No specific permission is required to store this information. 
```

## Direct Edit

> When performing a Direct Edit on a document, the document is stored in a temp folder inside `.nuxeo-drive`, no specific permission are required on this folder to properly use Direct Edit. 

Reviewer 2: ```suggestion
When performing a Direct Edit on a document, the document is stored in a temp folder inside `.nuxeo-drive`, no specific permission is required on this folder to use Direct Edit. 
```

> ### File Size

Reviewer 2: ```suggestion
### File Size Limits For Direct Edit
```

> To properly use Direct Edit please follow the recommendation of a document **up to 15GB file size with 65GB of RAM**. 

Reviewer 2: ```suggestion
Direct Edit has been tested with files **up to 15GB file size, with a machine having 65GB of RAM**. 
```

### After Direct Editing 

The document will not be deleted immediately after the direct edit process. The document is stored inside the `edit` folder under `.nuxeo-drive` folder. The folder will be cleaned at the time of Drive startup. So, until Nuxeo drive will be restarted, all of the documents will be kept locally.

**Recommended Storage Capacity:**

> As the documents are kept locally until Drive is restarted, it will depends on the number and size of the documents edited and how often the Nuxeo Drive gets restarted.

Reviewer 2: ```suggestion
As the documents are kept locally until Drive is restarted, the required storage size will depend on the number and size of the documents edited and how often the Nuxeo Drive gets restarted.
```

> To keep the storage free, Nuxeo Drive should be restarted on a regular basis if multiple direct editing are performed.

Reviewer 2: ```suggestion
To optimize storage space, Nuxeo Drive should be restarted on a regular basis if multiple direct editings are performed.
```

> ### Multiple Direct Editing 

Reviewer 2: This section is a repeat of previous information, could be deleted.

If the user edits multiple documents on a regular basis, it will increase the storage and memory usage.
The documents will be stored locally until Drive is restarted, the recommended free space should be at least twice the size of the document(s) to be edited. 

## Debugging 

> 

Reviewer 2: ```suggestion
When encountering issues with Nuxeo Drive, information is necessary to us to reproduce the issue and provide a quick and efficient resolution. We recommend following these steps:
```

1. The first thing to do is to change the log level to **DEBUG**. To do so, on the Advanced tab, go to Advanced Settings > Log level, select DEBUG from the drop-down list and click **Apply**. 

2. Then perform the whole operation you're having trouble with, and then generate the bug report. 

3. Apart from this we require:
- Nuxeo drive version
- System info: system configuration, RAM, free storage space, free memory space, CPU usage, system logs, server logs.  

> - Scenario/ steps to reproduce which resulted into the crash and information on the documents you are trying to edit (e.g: format, size), if a doc example can be shared it will be easier for the team to test on their side.

Reviewer 2: ```suggestion
- Scenario/ steps to reproduce which resulted into the crash and information on the documents you are trying to edit (e.g: format, size). 

Providing an example document greatly facilitates reproducing the issue. We understand that your content is sensitive and recommend that you share a version deprived of any sensitive information or personally identifying information as a precaution measure.
```
